datajoint/fetch.py

```diff
@@ -20,46 +25,57 @@ def prepare_attributes(relation, item):
         raise DataJointError("Index must be a slice, a tuple, a list, a string.")
     return item, attributes
 
-class FetchQuery:
+def copy_first(f):
+    @wraps(f)
+    def ret(*args, **kwargs):
+        args = list(args)
+        args[0] = args[0].__class__(args[0]) # call copy constructor
+        return f(*args, **kwargs)
 
-    def __init__(self, relation):
-        """
+    return ret
 
-        """
-        self.behavior = dict(
-            offset=0, limit=None, order_by=None, descending=False, as_dict=False, map=None
-        )
-        self._relation = relation
+class Fetch:
+    def __init__(self, relation):
+        if isinstance(relation, Fetch): # copy constructor
+            self.behavior = dict(relation.behavior)
+            self._relation = relation._relation
+        else:
+            self.behavior = dict(
+                offset=0, limit=None, order_by=None, as_dict=False
+            )
+            self._relation = relation
 
 
+    @copy_first
     def from_to(self, fro, to):
         self.behavior['offset'] = fro
         self.behavior['limit'] = to - fro
         return self
 
-    def order_by(self, order_by):
-        self.behavior['order_by'] = order_by
+    @copy_first
+    def order_by(self, *args):
+        if len(args) > 0:
+            self.behavior['order_by'] = self.behavior['order_by'] if self.behavior['order_by'] is not None else []
+            namepat = re.compile(r"\s*(?P<name>\w+).*")
+            for a in args: # remove duplicates
+                name = namepat.match(a).group('name')
+                pat = re.compile(r"%s(\s*$|\s+(\S*\s*)*$)" % (name,))
+                self.behavior['order_by'] = [e for e in self.behavior['order_by'] if not pat.match(e)]
+            self.behavior['order_by'].extend(args)
         return self
 
+    @copy_first
     def as_dict(self):
         self.behavior['as_dict'] = True
-
-    def ascending(self):
-        self.behavior['descending'] = False
         return self
 
-    def descending(self):
-        self.behavior['descending'] = True
-        return self
-
-    def apply(self, f):
-        self.behavior['map'] = f
-        return self
 
-    def limit_by(self, limit):
+    @copy_first
+    def limit_to(self, limit):
```

Comment:
It's clear that we have different opinions on this. Let's have a vote and stick with what the majority prefers. I suggest that the two options are 1. `limit`, `order`, `offset` and 2. `limit_to`, `order_by`, `offset_by`. I am happy to setup a doodle and send it around.